game-app/game-core/src/main/java/games/strategy/triplea/ui/panels/map/MapPanel.java

```diff
@@ -625,7 +625,12 @@ public void setGameData(final GameData data) {
     gameData.addTerritoryListener(territoryListener);
     gameData.addDataChangeListener(dataChangeListener);
     clearPendingDrawOperations();
-    executor.execute(() -> tileManager.resetTiles(gameData, uiContext.getMapData()));
+    // Try to mitigate race condition where game data is set after shutting down the executor
+    // technically there's no guarantee because the executor can be shut down after the if check
+    // and before the call to execute, but it should be good enough!
+    if (!executor.isTerminated()) {
```

Comment:
Any idea how we get into this method with a terminated executor?

I was wondering if it was some sort of cleanup and we were passing in a `null` `GameData` object to this method. Though.. looking at it, the null check seems nonsenical, if `data` were ever null we would get a NPE just a few lines earlier.